app/sagas/selectServer.ts

```diff
@@ -16,75 +24,127 @@ import log, { logServerVersion } from '../lib/methods/helpers/log';
 import I18n from '../i18n';
 import { BASIC_AUTH_KEY, setBasicAuth } from '../lib/methods/helpers/fetch';
 import { appStart } from '../actions/app';
+import { setSupportedVersions } from '../actions/supportedVersions';
 import UserPreferences from '../lib/methods/userPreferences';
 import { encryptionStop } from '../actions/encryption';
 import SSLPinning from '../lib/methods/helpers/sslPinning';
 import { inquiryReset } from '../ee/omnichannel/actions/inquiry';
-import { RootEnum } from '../definitions';
+import { IServerInfo, RootEnum, TServerModel } from '../definitions';
 import { CERTIFICATE_KEY, CURRENT_SERVER, TOKEN_KEY } from '../lib/constants';
-import { getLoginSettings, setCustomEmojis, setEnterpriseModules, setPermissions, setRoles, setSettings } from '../lib/methods';
+import {
+	checkSupportedVersions,
+	getLoginSettings,
+	setCustomEmojis,
+	setEnterpriseModules,
+	setPermissions,
+	setRoles,
+	setSettings
+} from '../lib/methods';
 import { Services } from '../lib/services';
 import { connect } from '../lib/services/connect';
+import { appSelector } from '../lib/hooks';
+import { getServerById } from '../lib/database/services/Server';
+import { getLoggedUserById } from '../lib/database/services/LoggedUser';
+
+const getServerVersion = function (version: string | null) {
+	let validVersion = valid(version);
+	if (validVersion) {
+		return validVersion;
+	}
+	const coercedVersion = coerce(version);
+	if (coercedVersion) {
+		validVersion = valid(coercedVersion);
+	}
+	if (validVersion) {
+		return validVersion;
+	}
+	throw new Error('Server version not found');
```

Comment:
This would be to generate the logs in bugsnag, right?
But will this information alone be enough to come to any conclusions?

Reply:
It should never get there. It's just theoretically possible, since semver throws.
If we find any issues, then we can improve it, if needed.
Unless you have anything in mind.